### ISSUE TITLE
What server to use?

### ISSUE THREAD

**Issue by Hairo** [szclsya/3DavSync#1]
When trying to use this with a [rclone webdav server](https://rclone.org/commands/rclone_serve_webdav/), i get the `Malformed WebDAV response: missing d:href in PROPFIND` error.

**Comment by szclsya** [comment]
I developed and tested this app with a Nextcloud WebDAV server, so it should work with that.

**Comment by Hairo** [comment]
Yeah, seems it only works with nextcloud webdav server, even tried with standalone apache (nextcould uses apache) and it didn't work.

**Comment by szclsya** [comment]
Looks like a WebDAV namespace issue. I don't have a 3DS nearby right now, but I guess I can push a fix if you want to test it.

**Comment by Hairo** [comment]
Sure, setting up a nextcloud server just for this is kinda overkill (though i did set it up to test that it only worked with NC).

**Comment by szclsya** [comment]
Try non-nextcloud-server branch? There's a new config to enable nextcloud style webdav namespace (`Nextcloud=true` under server setup), or use the RFC namespace (which should be more universally supported).

**Comment by Hairo** [comment]
Nice, working fine now with rclone, thank you.

**Comment by mgoeppl** [comment]
I'm currently running NC 29.0.4 and have the same problem: missing d:href in PROPFIND. failed to fetch remote file list...

The d:href is in the file when being downloaded using curl:

```
curl -u <username_redacted>:<password_redacted> -X PROPFIND -H "Depth: 1" https://<redacted>/remote.php/dav/files/<username_redacted>/3DS/3ds
```
```
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns" xmlns:nc="http://nextcloud.org/ns"><d:response><d:href>/remote.php/dav/files/<username_redacted>/3DS/3ds/</d:href><d:propstat><d:prop><d:getlastmodified>Mon, 05 Aug 2024 16:27:40 GMT</d:getlastmodified><d:resourcetype><d:collection/></d:resourcetype><d:quota-used-bytes>12292</d:quota-used-bytes><d:quota-available-bytes>-3</d:quota-available-bytes><d:getetag>&quot;66b0fd7c47fa0&quot;</d:getetag></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response><d:response><d:href>/remote.php/dav/files/<username_redacted>/3DS/3ds/.DS_Store</d:href><d:propstat><d:prop><d:getlastmodified>Sun, 04 Aug 2024 22:36:58 GMT</d:getlastmodified><d:getcontentlength>8196</d:getcontentlength><d:resourcetype/><d:getetag>&quot;9a084ee29921b9c96567f9e3ad7c89d9&quot;</d:getetag><d:getcontenttype>application/octet-stream</d:getcontenttype></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response><d:response><d:href>/remote.php/dav/files/<username_redacted>/3DS/3ds/._.DS_Store</d:href><d:propstat><d:prop><d:getlastmodified>Sun, 04 Aug 2024 22:34:49 GMT</d:getlastmodified><d:getcontentlength>4096</d:getcontentlength><d:resourcetype/><d:getetag>&quot;c7b07f580e7cca49b10cdc9add2a12af&quot;</d:getetag><d:getcontenttype>application/octet-stream</d:getcontenttype></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response></d:multistatus>
```
My config is as follows:
```
[General]
# List webdav configs that will be synced
Enabled=3ds

[3ds]
Url=https://<redacted>/remote.php/dav/files/<redacted>/3DS/
LocalPath=/3ds
# Specify credential here
Username=<redacted>
Password=<redacted>
Nextcloud=true
```
